src/Neo/SmartContract/Native/GasToken.cs

```diff
@@ -36,9 +36,11 @@ internal override async ContractTask OnPersist(ApplicationEngine engine)
             long totalNetworkFee = 0;
             foreach (Transaction tx in engine.PersistingBlock.Transactions)
             {
-                await Burn(engine, tx.Sender, tx.SystemFee + tx.NetworkFee);
+                var gasConsumed = engine.PersistingBlock._gasConsumed[tx.Hash];
```

Comment:
Isn't it always zero here?

Reply:
No, it will be initialied while persisting blocks

Reply:
> Isn't it always zero here?

That is why i need to study on this T_T~~~kinda more complex than thought to achieve this function.

Reply:
8112bd6e61f2d7a0ce0361931ef6d76c9b0d61a4 unfortunately breaks `OnPersist`, it MUST be run before transactions. Suppose I have 1 GAS on my account and I send a transaction that sends 1 GAS to someone with 1 GAS fees. Normally I'd have 1 GAS written off for fees in `OnPersist` and then the transaction will fail because of insufficient balance (which is expected and OK). With your change I'd have 1 GAS sent to another account and then `OnPersist` will fail because of insufficient balance, but that will just ruin the chain, `OnPersist` can't fail.

Thanks for the statistics, btw! 85%+ with 90%+ GAS consumed is not that bad, but I fear if we map it onto the blockchain history we'd have more "inefficient" transactions now than we've had a year ago.

Reply:
I am just trying to find a solution as simple as possible~~~still thinking on that......as to the gas fee, indeed, DEFI contracts like FlamingGo are more likely to have transactions pay more fees.